src/adapter/src/coord.rs

```diff
@@ -604,6 +603,22 @@ impl<S: Append + 'static> Coordinator<S> {
                 entry.item().typ(),
                 entry.id()
             );
+
+            // For things that need the storage collection to exist, sync all
+            // queued collections.
+            if matches!(
+                entry.item(),
+                CatalogItem::Index(_) | CatalogItem::View(_) | CatalogItem::MaterializedView(_)
+            ) && !collections_to_create.is_empty()
+            {
+                let collections = collections_to_create.drain(..).collect();
+                self.controller
+                    .storage
+                    .create_collections(collections)
+                    .await
+                    .unwrap();
+            }
```

Comment:
It might be worth doing some sorting so that we can guarantee that all sources and tables come first. That would maximize the batching done.

Reply:
Also I remember from the last time we tried this that there was a reason, but I just want to double check. Why can't we just send all of the storage collections as one giant batch, and then send all of the dataflows as one giant batch? Is it because a materialized view storage collection may depend on some compute dataflow?